### PR TITLE
Add link() in build.zig for being used as dependency

### PR DESCRIPTION
**Pull request by iacore**
P.S. zig-libressl doesn't build on my machine (Arch Linux) unless I use the "use system library" option.

On Arch Linux it's as easy as `pacman -S libressl` to install LibreSSL.

### CODE REVIEW

**Comment by haze** [comment]
@locriacyber Do you mind giving me logs on what isn't building? I've tested and made sure Zig can build a checked out libressl on both macOS and popOS. I'm not a heavy linux user so I'm not sure if there would be a difference between the two that I didn't acknowledge.

Anyways, thanks for the PR!

File: build.zig [hunk 1]
```diff
@@ -1,10 +1,11 @@
 const std = @import("std");
-const zig_libressl = @import("zig-libressl/build.zig");
+pub const zig_libressl = @import("zig-libressl/build.zig");
 const Pkg = std.build.Pkg;
 
 fn relativeToThis(comptime path: []const u8) []const u8 {
     comptime {
-        return std.fs.path.dirname(@src().file).? ++ "/" ++ path;
+        const fspath = std.fs.path;
+        return fspath.dirname(@src().file).? ++ fspath.sep_str ++ path;
```
Comment (haze) [inline build.zig:8]: In the future I'd like to avoid local shortenings like this

**Review by haze — APPROVED** [review]

File: build.zig (hunk 1 above)
Comment (iacore) [reply build.zig:8]: You mean writing out the full path? `std.fs.path` is long.

Comment (haze) [reply build.zig:8]: Yes

**Pull request merged**